include/spdk/bdev.h

```diff
@@ -891,6 +891,73 @@ int spdk_bdev_readv_blocks_with_md(struct spdk_bdev_desc *desc, struct spdk_io_c
 				   uint64_t offset_blocks, uint64_t num_blocks,
 				   spdk_bdev_io_completion_cb cb, void *cb_arg);
 
+/**
+ * The type of context which is used in \b spdk_bdev_io_memory_key_ctx
+ */
+enum spdk_bdev_io_memory_key_ctx_type {
+	/** Context of this type points to ibv_pd structure */
+	SPDK_BDEV_MEMORY_KEY_CTX_PD = 0
+};
+
+struct spdk_bdev_io_memory_key_ctx {
+	void *ctx;
+	enum spdk_bdev_io_memory_key_ctx_type ctx_type;
+	/** Memory key to be filled by the user of \b spdk_bdev_readv_with_md_ext of \b spdk_bdev_writev_with_md_ext
+	 * when bdev module calls \b spdk_bdev_io_get_mkey */
+	uint32_t mkey;
```

Comment:
Need more description here. It is not very clear that ctx and ctx_typre are input parameters and mkey is out parameter.